iothub/device/src/ConnectionInfo.cs

```diff
@@ -44,5 +46,50 @@ internal ConnectionInfo(ConnectionStatus status, ConnectionStatusChangeReason ch
         /// Timestamp in UTC when the last connection status was changed.
         /// </summary>
         public DateTimeOffset StatusLastChangedOnUtc { get; internal set; }
+
+        /// <summary>
+        /// Recommended actions for users to take upon different ConnectionStatus and ConnectionStatusChangeReason.
+        /// </summary>
+        /// <remark>
+        /// Defaults to <see cref="RecommendedAction.DisposeAndOpenIfWish"/>.
+        /// </remark>>
+        public RecommendedAction RecommendedAction { get; internal set; }
+
+        private RecommendedAction GetRecommendedActionBasedOnConnectionStatusAndChangeReason(ConnectionStatus status, ConnectionStatusChangeReason changeReason)
```

Comment:
This is the mapping between <<`ConnectionStatus`, `ConnectionStatusChangeReason`>, `RecommendedAction`>